qemu/tests/numa_node_affinity.py

```diff
@@ -0,0 +1,62 @@
+import re
+
+from virttest import env_process
+from virttest import error_context
+from virttest import utils_misc
+
+from virttest.qemu_monitor import QMPCmdError
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    numa_node_affinity test
+    1) Check the NUMA topology, cancel if there is a single node.
+    2) Boot up a guest with node-affinity selecting the first valid node.
+    3) Check the cpu-affinity obtained from QEMU is correct.
+    4) Check the node-affinity property is not readable.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    host_numa_node = utils_misc.NumaInfo()
+    node_list = host_numa_node.online_nodes_withcpu
+    if len(node_list) < 2:
+        test.cancel("Host only has one NUMA node, skipping test...")
```

Comment:
Could you explain more about the reason of using 2 valid node at least ? Thanks!

Reply:
@yanan-fu in fact one valid node should be enough, but it's interesting to check how this property works with more than one node because that's the truly intention of it. Also that's the reason of the condition from `L48-49`, it's important to test there is no bad/cross CPU assignment between nodes.

Reply:
Make sense, thanks!